From e7eca00d90d00ba8218f061a6f52d367c26c5f4a Mon Sep 17 00:00:00 2001
From: Madhu Khedar <madhu.khedar@acrolinx.com>
Date: Tue, 21 Feb 2017 15:57:26 +0530
Subject: [PATCH 2/2] DEV-10276 index.html always reloaded

---
 src/utils/sidebar-loader.ts | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/src/utils/sidebar-loader.ts b/src/utils/sidebar-loader.ts
index d435209..40433e7 100644
--- a/src/utils/sidebar-loader.ts
+++ b/src/utils/sidebar-loader.ts
@@ -33,6 +33,11 @@ function createScriptElement(src: string) {
   return el;
 }
 
+function createCompleteSidebarUrl(sidebarBaseUrl: string) {
+	const timestamp = Date.now();
+	return sidebarBaseUrl + 'index.html?t=' + timestamp;
+}
+
 /**
  * Loads the Styles and Scripts of the sidebar into the current window.
  * @param sidebarUrl must end with /
@@ -41,8 +46,9 @@ export function loadSidebarCode(sidebarUrl = SIDEBAR_URL) {
   const sidebarBaseUrl = sidebarUrl;
 
   const getAbsoluteAttributeValue = (s: string) => s.replace(/^.*"(.*)".*$/g, sidebarBaseUrl + '$1');
-
-  utils.fetch(sidebarBaseUrl + 'index.html', sidebarHtml => {
+  
+  const completeSidebarUrl = createCompleteSidebarUrl(sidebarBaseUrl);
+  utils.fetch(completeSidebarUrl, sidebarHtml => {
     if (sidebarHtml.indexOf("<meta name=\"sidebar-version\"") < 0) {
       try {
         throw new SidebarURLInvalidError("It looks like the sidebar URL was configured wrongly.", sidebarBaseUrl, sidebarHtml);
@@ -70,7 +76,7 @@ export function loadSidebarCode(sidebarUrl = SIDEBAR_URL) {
 
 export function loadSidebarIntoIFrame(config: AcrolinxPluginConfig, sidebarIFrameElement: HTMLIFrameElement, onSidebarLoaded: () => void) {
   const sidebarBaseUrl = config.sidebarUrl || SIDEBAR_URL;
-  const completeSidebarUrl = sidebarBaseUrl + 'index.html';
+  const completeSidebarUrl = createCompleteSidebarUrl(sidebarBaseUrl);
   if (config.useMessageAdapter || (config.useSidebarFromSameOriginDirectly && utils.isFromSameOrigin(sidebarBaseUrl))) {
     sidebarIFrameElement.addEventListener('load', onSidebarLoaded);
     sidebarIFrameElement.src = completeSidebarUrl;
-- 
2.10.0.windows.1

